edit_book.php

```diff
@@ -0,0 +1,44 @@
+<?php
+include 'database.php';
```

Comment:
**Observations**: The SQL query construction uses user input directly, which can lead to SQL injection vulnerabilities.

**Recommendations**: Use prepared statements for the SELECT and UPDATE queries. Here’s an example for the UPDATE operation:

```php
if ($_SERVER['REQUEST_METHOD'] == 'POST') {
    $stmt = $connection->prepare("UPDATE books SET title = ?, author = ?, year = ? WHERE id = ?");
    $stmt->bind_param("ssii", $title, $author, $year, $id);
    $stmt->execute();
    if ($stmt->affected_rows > 0) {
        echo "Book updated successfully!";
        header("Location: list_books.php");
        exit();
    } else {
        echo "Error: " . $stmt->error;
    }
    $stmt->close();
}
```